src/header/common/link.rs

```diff
@@ -0,0 +1,1027 @@
+
+        self
+    }
+
+    /// Add a `RelationType` to the `LinkValue`'s `rev` parameter.
+    pub fn push_rev(mut self, rev: RelationType) -> LinkValue {
+        let mut v = self.rev.take().unwrap_or(Vec::new());
+
+        v.push(rev);
+
+        self.rev = Some(v);
+
+        self
+    }
+
+    /// Add a `LanguageTag` to the `LinkValue`'s `hreflang` parameter.
+    pub fn push_href_lang(mut self, language_tag: LanguageTag) -> LinkValue {
+        let mut v = self.href_lang.take().unwrap_or(Vec::new());
+
+        v.push(language_tag);
+
+        self.href_lang = Some(v);
+
+        self
+    }
+
+    /// Add a `MediaDesc` to the `LinkValue`'s `media_desc` parameter.
+    pub fn push_media_desc(mut self, media_desc: MediaDesc) -> LinkValue {
+        let mut v = self.media_desc.take().unwrap_or(Vec::new());
+
+        v.push(media_desc);
+
+        self.media_desc = Some(v);
+
+        self
+    }
+
+    /// Set `LinkValue`'s `title` parameter.
+    pub fn set_title(mut self, title: &str) -> LinkValue {
+        self.title = Some(String::from(title));
+
+        self
+    }
+
+    /// Set `LinkValue`'s `title*` parameter.
+    pub fn set_title_star(mut self, title_star: &str) -> LinkValue {
+        self.title_star = Some(String::from(title_star));
+
+        self
+    }
+
+    /// Set `LinkValue`'s `type` parameter.
+    pub fn set_media_type(mut self, media_type: &Mime) -> LinkValue {
+        self.media_type = Some(media_type.clone());
+
+        self
+    }
+
+    /// Set `LinkValue`'s `link-extension` parameter.
+    pub fn set_link_extension(mut self, link_extension: &str) -> LinkValue {
+        self.link_extension = Some(String::from(link_extension));
+
+        self
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Trait implementations
+////////////////////////////////////////////////////////////////////////////////
+
+impl Header for Link {
+    fn header_name() -> &'static str {
+        static NAME: &'static str = "Link";
+        NAME
+    }
+
+    fn parse_header(raw: &Raw) -> ::Result<Link> {
+        // If more that one `Link` headers are present in a request's
+        // headers they are combined in a single `Link` header containing
+        // all the `link-value`s present in each of those `Link` headers.
+        raw.iter()
+            .map(parsing::from_raw_str::<Link>)
+            .fold(None, |p, c| {
+                match (p, c) {
+                    (None, c) => Some(c),
+                    (e @ Some(Err(_)), _) => e,
+                    (Some(Ok(mut p)), Ok(c)) => {
+                        p.values.extend(c.values);
+
+                        Some(Ok(p))
+                    },
+                    _ => Some(Err(::Error::Header)),
+                }
+            })
+            .unwrap_or(Err(::Error::Header))
+    }
+
+    fn fmt_header(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt_delimited(f, self.values.as_slice(), ", ", "", "")
```

Comment:
I agree, it doesn't look very good. I changed it to receive a tuple, I think it's a bit better. I thought about creating a tuple struct and passing it but it won't be so efficient I believe.